### PR TITLE
Implement _mm_shuffle_epi8 with NEON instruction

### PR DESCRIPTION
**Pull request by marktwtn**
Add the corresponding unit testing.

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -1000,33 +1000,13 @@ FORCE_INLINE __m128i _mm_shuffle_epi_3332(__m128i a)
...
 // corresponding 8-bit element of b, and store the results in dst.
-FORCE_INLINE __m128i _mm_shuffle_epi8(__m128i ia, __m128i ib)
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_shuffle_epi8&expand=5146
+FORCE_INLINE __m128i _mm_shuffle_epi8(__m128i a, __m128i b)
 {
-    uint8_t *a = (uint8_t *) &ia;  // input a
-    uint8_t *b = (uint8_t *) &ib;  // input b
-    int32_t r[4];
+    int8x16_t t = vreinterpretq_s8_m128i(a);     // input a
+    uint8x16_t idx = vreinterpretq_u8_m128i(b);  // input b
```
Comment (jserv) [inline sse2neon.h]: Can you replace `t` and `idx` with something mnemonic?

**Comment by jserv** [comment]
Please append corresponding test case.

File: sse2neon.h (hunk 1 above)
Comment (marktwtn) [reply sse2neon.h]: I just follow the input argument naming by searching the `vqtbl1q_s8` in the [Neon intrinsic page](https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics).

I think the `t` variable can be changed to `tbl`, or remains the same to follow the description in the Neon intrinsic page.

Comment (jserv) [reply sse2neon.h]: `tbl` sounds better since it is a matter of table lookup operations.

Comment (marktwtn) [reply sse2neon.h]: Done.

**Comment by marktwtn** [comment]
I have added the corresponding unit testing and modified the implementation of `_mm_shuffle_epi8`.
The modification is based on the difference behaviour of `_mm_shuffle_epi8` and `vqtbl1q_s8`.

However, the unit testing is still not passing.
The pull request should not be merged until I fix the bug.

**Comment by marktwtn** [comment]
The pull request can be merged without problem now.
The unit testing has been fixed.
I wrote the wrong unit testing since I forgot the difference of the signed/unsigned integer casting.

**Pull request merged**